### ISSUE TITLE
[bug]:  Wrong selector for select component placeholder

### ISSUE THREAD

**Issue by yudyananda** [shadcn-ui/ui#3737]
### Describe the bug

Just found out that tailwind selector `placeholder:...` on trigger (Select component) won't have any effect. In my case`data-[placeholder]` will solve the issue

### Affected component/components

Select

### How to reproduce

change the style of the trigger placeholder in the select component, and see if there is a change

### System Info

```bash
WSL
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by erashu212** [comment]
@yudyananda Can you please share any codesandbox?

**Comment by servesh-chaturvedi** [comment]
The tailwind placeholder selector is not working for you as the styles will only affect an input component where placeholder is an attribute. On the select component it is used as a prop so it won't work. I don't think we can call this a bug @yudyananda

**Comment by abeisleem** [comment]
@yudyananda is right. see documentation here: https://www.radix-ui.com/primitives/docs/components/select#with-a-placeholder

currently, the select component is styling the placeholder text with `placeholder:text-muted-foreground` and based on radix select component, it should be styled with `data-[placeholder]:text-muted-foreground`. the current `placeholder:text-muted-foreground` does nothing to the style. 

applying the correct placeholder selection has impact on the color of the ChevronDown in making it also `text-muted-foreground`... to resolve this, it's should color should explicitly be styled with `text-foreground`

see the following differences: 

with the current placeholder selector: 
![image](https://github.com/shadcn-ui/ui/assets/29489334/f31015bc-2a99-4170-ba07-7f1b3baea114)

with the correct placeholder selector: 
![image](https://github.com/shadcn-ui/ui/assets/29489334/0d707b2b-4c0a-4406-b3e7-5517e3d0c4c3)

with the correct placeholder selector AND explicitly styling the color of the chevron: 
![image](https://github.com/shadcn-ui/ui/assets/29489334/19ce8e3d-3017-4744-b0b1-3031458e6fb5)

**Comment by yudyananda** [comment]
> The tailwind placeholder selector is not working for you as the styles will only affect an input/textarea element where placeholder is an attribute. On the select component it is used as a prop so it won't work. @yudyananda

@servesh-chaturvedi  Yes i'm on the same page with you about that, but just like @abeisleem said  the current selector `placeholder:...` on the select component does nothing because the trigger it self is only a button who act as pseudo element for the hidden input

**Comment by adham618** [comment]
Why not merged yet?

**Comment by IPJT** [comment]
> [@yudyananda](https://github.com/yudyananda) is right. see documentation here: https://www.radix-ui.com/primitives/docs/components/select#with-a-placeholder
> 
> currently, the select component is styling the placeholder text with `placeholder:text-muted-foreground` and based on radix select component, it should be styled with `data-[placeholder]:text-muted-foreground`. the current `placeholder:text-muted-foreground` does nothing to the style.
> 
> applying the correct placeholder selection has impact on the color of the ChevronDown in making it also `text-muted-foreground`... to resolve this, it's should color should explicitly be styled with `text-foreground`
> 
> see the following differences:
> 
> with the current placeholder selector: ![image](https://github.com/shadcn-ui/ui/assets/29489334/f31015bc-2a99-4170-ba07-7f1b3baea114)
> 
> with the correct placeholder selector: ![image](https://github.com/shadcn-ui/ui/assets/29489334/0d707b2b-4c0a-4406-b3e7-5517e3d0c4c3)
> 
> with the correct placeholder selector AND explicitly styling the color of the chevron: ![image](https://github.com/shadcn-ui/ui/assets/29489334/19ce8e3d-3017-4744-b0b1-3031458e6fb5)
> 
> Edit: there's currently an outstanding PR ([#2558](https://github.com/shadcn-ui/ui/pull/2558)) for this

@abeisleem - Thanks for pointing this out. Wouldn't have noticed as it was quite subtle. 

Rather than setting the color of the chevron to `text-foreground` I'd rather set it to `text-inherit` so that it gets the correct color regardless of how its parents are styled:

```JS
<ChevronDownIcon className="h-4 w-4 text-inherit opacity-50" />
```

**Comment by shadcn** [comment]
Hi. We're closing some old issues as outdated. If this is still relevant, leave a message, we'll reopen it. Thank you. Appreciate your contribution to the project - shadcn
